app/Http/Middleware/EnsureUserHasRole.php

```
@@ -0,0 +1,32 @@
+
+    //Middleware can take extra arguments.
+    public function handle(Request $request, Closure $next, string $role): Response
+    {
+        if (! $request->user()->hasRole($role)) {
+            return redirect('/');
+        }
+        return $next($request);
+    }
+
+    // middleware to work after the response has been sent (like logging).
+    public function terminate(Request $request, Response $response)
```

PHPMD (2.15.0)

Avoid unused parameters such as '$request'. (undefined)

(UnusedFormalParameter)